web/ui/react-app/src/Graph.tsx

```diff
@@ -217,25 +237,25 @@ class Graph extends PureComponent<GraphProps> {
     return val;
   }
 
-  componentDidMount() {
-    this.plot();
-  }
-
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: GraphProps) {
+    if (prevProps.data !== this.props.data) {
+      this.setState({ selectedSeriesIndex: -1 });
+    }
     this.plot();
   }
 
   componentWillUnmount() {
     this.destroyPlot();
   }
 
-  plot() {
-    if (this.chartRef.current === null) {
+  plot = () => {
+    if (!this.chartRef.current) {
       return;
     }
+    const selectedData = this.getData()[this.state.selectedSeriesIndex];
     this.destroyPlot();
-    $.plot($(this.chartRef.current!), this.getData(), this.getOptions());
-  }
+    $.plot($(this.chartRef.current), selectedData ? [selectedData] : this.getData(), this.getOptions());
```

Comment:
By just plotting the selected series like this, the Y axis range changes to the range of the single selected series. In the old UI, the Y axis didn't update. I think either behavior can have an advantage at different times (e.g. zooming in into a small series vs. not having the Y axis jump around so you don't lose your orientation).

I'm not sure yet what to do here, but we might also want to keep the old behavior here to keep the graph range fixed to the entire data range, independent of individual series visibility.

Reply:
Personnally the old behaviour always annoyed me :)

Reply:
@roidelapluie Thanks, that's the feedback I was looking for :) I also agree, let's just keep it with the new behavior for now, which is likely actually going to be more useful for people. And if people really complain, we can see.